web/models/package_metadata.ex

```diff
@@ -12,5 +12,22 @@ defmodule HexWeb.PackageMetadata do
   def changeset(meta, params) do
     cast(meta, params, ~w(description licenses links maintainers extra))
     |> validate_required(~w(description licenses)a)
+    |> validate_links()
+  end
+
+  defp validate_links(changeset) do
+    validate_change(changeset, :links, fn _, links ->
+      links
+      |> Map.values()
+      |> Enum.reject(&valid_url?/1)
+      |> Enum.map(&{:links, "invalid link #{inspect &1}"})
+    end)
+  end
+
+  defp valid_url?(url) do
+    case :http_uri.parse(to_charlist(url)) do
```

Comment:
this only allows HTTP(s) urls, but looking at existing links people tend to use only those (https://gist.github.com/wojtekmach/de087da39ecc62ea41c4367db990cd88)